rs/boundary_node/rate_limits/canister/fetcher.rs

```diff
@@ -101,43 +122,102 @@ impl<R: CanisterApi, F: ConfidentialityFormatting<Input = OutputConfig>> EntityF
         };
 
         Ok(config)
     }
 }
 
-impl<R: CanisterApi, F: ConfidentialityFormatting<Input = OutputRuleMetadata>> EntityFetcher
-    for RuleFetcher<R, F>
+impl<
+        R: CanisterApi,
+        F: ConfidentialityFormatting<Input = OutputRuleMetadata>,
+        A: ResolveAccessLevel,
+    > EntityFetcher for IncidentFetcher<R, F, A>
+{
+    type Input = rate_limits_api::IncidentId;
+    type Output = Vec<rate_limits_api::OutputRuleMetadata>;
+    type Error = FetchError;
+
+    fn fetch(&self, incident_id: Self::Input) -> Result<Self::Output, Self::Error> {
+        let incident_id = IncidentId::try_from(incident_id.clone())
+            .map_err(|_| FetchError::InvalidUuidFormat(incident_id))?;
+
+        let incident_metadata = self
+            .canister_api
+            .get_incident(&incident_id)
+            .ok_or_else(|| FetchError::NotFound(incident_id.0.to_string()))?;
+
+        let mut output_rules = Vec::with_capacity(incident_metadata.rule_ids.len());
+
+        for rule_id in incident_metadata.rule_ids.into_iter() {
+            let stored_rule_metadata = self.canister_api.get_rule(&rule_id).ok_or_else(|| {
+                FetchError::UnexpectedError(anyhow::anyhow!("Rule with id = {rule_id} not found"))
+            })?;
+
+            let mut rule_metadata = OutputRuleMetadata {
+                id: rule_id,
+                incident_id,
+                rule_raw: Some(stored_rule_metadata.rule_raw),
+                description: Some(stored_rule_metadata.description),
+                disclosed_at: stored_rule_metadata.disclosed_at,
+                added_in_version: stored_rule_metadata.added_in_version,
+                removed_in_version: stored_rule_metadata.removed_in_version,
+            };
+
+            // If the rule is not disclosed and the viewer is not authorized, redact the rule.
+            if rule_metadata.disclosed_at.is_none() {
+                let is_authorized_viewer = self.access_resolver.get_access_level()
+                    == AccessLevel::FullAccess
+                    || self.access_resolver.get_access_level() == AccessLevel::FullRead;
+                if !is_authorized_viewer {
+                    rule_metadata = self.formatter.format(&rule_metadata);
+                }
+            }
+
+            output_rules.push(rule_metadata.into());
+        }
+
+        Ok(output_rules)
+    }
+}
+
+impl<
+        R: CanisterApi,
+        F: ConfidentialityFormatting<Input = OutputRuleMetadata>,
+        A: ResolveAccessLevel,
+    > EntityFetcher for RuleFetcher<R, F, A>
 {
     type Input = rate_limits_api::RuleId;
     type Output = rate_limits_api::OutputRuleMetadata;
-    type Error = FetchRuleError;
+    type Error = FetchError;
 
-    fn fetch(
-        &self,
-        rule_id: rate_limits_api::RuleId,
-    ) -> Result<rate_limits_api::OutputRuleMetadata, FetchRuleError> {
+    fn fetch(&self, rule_id: rate_limits_api::RuleId) -> Result<Self::Output, Self::Error> {
```

Comment:
Why not the following to be consistent?
```suggestion
    fn fetch(&self, rule_id: Self::Input) -> Result<Self::Output, Self::Error> {
```